File: src/game/tools/rpc/output/wrap/BattleHeroWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class BattleHeroWrap : IWrapper
	{	
		public int HeroID; // 英雄ID
		public int Level; // 英雄等级
		
		public void Decode(Packet pck)
		{	
			HeroID = pck.GetInt(); 
			Level = pck.GetInt(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutInt(HeroID); 
        	pck.PutInt(Level); 
        }
	}
}

File: src/game/tools/rpc/output/wrap/BattleHeroWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type BattleHeroWrap struct {
	
	HeroID int32 // 英雄ID
	Level int32 // 英雄等级
	rpc.Wrapper
}

func (w *BattleHeroWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.HeroID = pck.PopInt32(); 
	w.Level = pck.PopInt32(); 
	return w
}

func (w *BattleHeroWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt32(w.HeroID) 
	pck.PutInt32(w.Level) 
}

